.github/workflows/go.yml

```
@@ -26,10 +32,27 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+        with:
+          submodules: recursive
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
           go-version-file: "go.mod"
+      - name: Download and extract execution spec tests
+        shell: bash
+        run: |
+          set -euxo pipefail
+          mkdir -p "${EXECUTION_SPEC_TESTS_PATH}"
+          # Download the archive
+          curl -L -o /tmp/fixtures_develop.tar.gz https://github.com/ethereum/execution-spec-tests/releases/download/${EXECUTION_SPEC_TESTS_VERSION}/fixtures_develop.tar.gz
+          # Verify checksum (replace EXPECTED_CHECKSUM with actual value)
+          ACTUAL_CHECKSUM=$(sha256sum /tmp/fixtures_develop.tar.gz | cut -d' ' -f1)
+          if [ "$ACTUAL_CHECKSUM" != "$EXECUTION_SPEC_TESTS_CHECKSUM" ]; then
+            echo "Checksum mismatch! Expected: $EXPECTED_CHECKSUM, Got: $ACTUAL_CHECKSUM"
+            exit 1
+          fi
+          # Extract if verification passes
+          tar -xz -f /tmp/fixtures_develop.tar.gz -C "${EXECUTION_SPEC_TESTS_PATH}"
```

```suggestion
- name: Download spec tests
        run: curl -o "${EXECUTION_SPEC_TESTS_FILE}" -L "https://github.com/ethereum/execution-spec-tests/releases/download/${EXECUTION_SPEC_TESTS_VERSION}/${EXECUTION_SPEC_TESTS_FILE}"
      - name: Verify spec tests
        run: |
          sha256sum --ignore-missing --check build/checksums.txt | grep "${EXECUTION_SPEC_TESTS_FILE}: OK"
      - name: Extract spec tests
        run: |
          mkdir -p tests/spec-tests && tar -xz -f "${EXECUTION_SPEC_TESTS_FILE}" -C tests/spec-tests
```

Note that this will also need `EXECUTION_SPEC_TESTS_FILE: fixtures_develop.tar.gz` in the job's `env`.

Rationale:

1. Using workflow steps removes the need for a full Bash script, is self-documenting (one of the comments was already out of sync with the command), and implicitly does the same as `set -e`. Both `pipefail` and `-u` are redundant (even though usually good practice IMO) because the commands would fail and trigger `-e`.
2. `sha256sum` provides functionality to check its output against known values so we shouldn't replicate it.